test/integration/component/test_escalation_listTemplateDomainAdmin.py

```diff
@@ -0,0 +1,157 @@
+        builtin_info = get_builtin_template_info(cls.apiclient, cls.zone.id)
+        cls.testdata["privatetemplate"]["url"] = builtin_info[0]
+        cls.testdata["privatetemplate"]["hypervisor"] = builtin_info[1]
+        cls.testdata["privatetemplate"]["format"] = builtin_info[2]
+        cls.cleanup = []
+
+# Create 2 domain admin accounts
+
+        cls.domain1 = Domain.create(
+            cls.apiclient,
+            cls.testdata["domain"])
+
+        cls.domain2 = Domain.create(
+            cls.apiclient,
+            cls.testdata["domain"])
+
+        cls.account1 = Account.create(
+            cls.apiclient,
+            cls.testdata["account"],
+            admin=True,
+            domainid=cls.domain1.id)
+
+        cls.account2 = Account.create(
+            cls.apiclient,
+            cls.testdata["account2"],
+            admin=True,
+            domainid=cls.domain2.id)
+
+        cls.debug("Created account %s in domain %s" %
+                  (cls.account1.name, cls.domain1.id))
+        cls.debug("Created account %s in domain %s" %
+                  (cls.account2.name, cls.domain2.id))
+
+        cls.cleanup.append(cls.account1)
+        cls.cleanup.append(cls.account2)
+        cls.cleanup.append(cls.domain1)
+        cls.cleanup.append(cls.domain2)
+
+    @classmethod
+    def tearDownClass(cls):
+        try:
+            # Cleanup resources used
+            cleanup_resources(cls.apiclient, cls.cleanup)
+
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+        return
+
+# test that the template register under root/domain1->account1 is not
+# listed under root/domain2->account2
+
+    @attr(tags=["advanced", "basic"], required_hardware="true")
+    def test_listtemplate(self):
+
+        # Register template under one domain admin(root/domain1->account 1)
+
+        template_register = Template.register(
+            self.apiclient,
+            self.testdata["privatetemplate"],
+            zoneid=self.zone.id,
+            hypervisor=self.hypervisor,
+            account=self.account1.name,
+            domainid=self.domain1.id)
+
+        template_register.download(self.apiclient)
+        # self.cleanup.append(self.template_register)
+
+        time.sleep(self.testdata["sleep"])
+        timeout = self.testdata["timeout"]
+        while True:
+            listTemplateResponse = Template.list(
+                self.apiclient,
+                templatefilter="all",
+                id=template_register.id,
+                account=self.account1.name,
+                domainid=self.domain1.id
+            )
+            if isinstance(listTemplateResponse, list):
+                break
+            elif timeout == 0:
+                raise Exception("list template failed")
+
+            time.sleep(10)
+            timeout = timeout - 1
+
+            self.assertEqual(
+                isinstance(listTemplateResponse, list),
```

Comment:
isinstance and len() check for list can be combined by using validateList method from common.py. Use only single method.